db/migrate/20230628044856_add_unique_to_sources.rb

```diff
@@ -0,0 +1,6 @@
+class AddUniqueToSources < ActiveRecord::Migration[7.0]
+  def change
+    change_column :sources, :name, :string, unique: true, null: false
+    add_index :sources, :name, unique: true
+  end
+end
```

Comment:
These migrations are quite dangerous if you do not have a "data migration" to run before.
In case your DB contains duplicated names, the migration will fail to run and so the application will not start.

Here is an example of migration from our current project:

```
# frozen_string_literal: true

class AddUniqueIndexOnNameToTenants < ActiveRecord::Migration[7.0]
  def up
    update_existing_tenants

    remove_index :tenants, :name
    add_index :tenants, :name, unique: true, where: 'discarded_at IS NULL'
  end

  def down
    remove_index :tenants, :name
    add_index :tenants, :name
  end

  private

  def update_existing_tenants
    Tenant.with_discarded.group_by { |tenant| tenant.name.downcase }
                         .each do |_name, tenants|
      next unless tenants.size > 1

      tenants.each_with_index do |tenant, index|
        tenant.name = "#{tenant.name} #{index + 1}"
        tenant.save!(validate: false)
      end
    end
  end
end
```

Reply:
I added the `update_existing` logic to the migration in https://github.com/nimblehq/ic-rails-phong-bliss/pull/65/commits/1dc37ad49806641f8310f8229f4724d24bbfbebc. I still use change because the index `name` did not exist before this migration.